tasks/tool.py

```diff
@@ -165,6 +176,25 @@ def is_linux():
     return platform.system() == "Linux"
 
 
+def is_wsl():
+    return "microsoft" in platform.uname().release.lower()
+
+
+def get_aws_instance_password_data(
+    ctx: Context, vm_id: str, key_path: str, aws_account: Optional[str] = None, use_aws_vault: Optional[bool] = True
+) -> str:
+    buffer = StringIO()
+    with ctx.cd(_get_root_path()):
+        cmd = f"aws ec2 get-password-data --instance-id {vm_id} --priv-launch-key {key_path}"
+        if use_aws_vault:
+            if aws_account is None:
+                raise Exit("AWS account is required when using aws-vault.")
+            cmd = get_aws_wrapper(aws_account) + cmd
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Should not be required since https://github.com/DataDog/test-infra-definitions/pull/917, as you call `aws` directly you might prepend with `AWS_PROFILE` and the default profile name for the current target from config, if `AWS_PROFILE` is not set yet

Something like
```
prefix = ""
if os.env("AWS_PROFILE") is None:
  prefix = `"AWS_PROFILE=exec-sso-agent-[sandbox|qa]-account-admin "`
```

The advantage is that you don't need to pass the `use_aws_vault` option

Reply:
discussed on slackL

will merge this PR as is, since the profile isn't used by python yet, and there are other places that still use `get_aws_wrapper`. They can all be changed together in a separate PR.